lib/libc/stdio/printf.3

```diff
@@ -653,43 +1034,112 @@ the number of bytes required to render the multibyte representation of
 the string, the array must contain a terminating wide
 .Dv NUL
 character.
-.It Cm p
+.It Cm s
+.Sm off
+.Cm %
+.Op Ar argno Cm $
+.Op Cm \-
+.Op Ar width
+.Op . Ar maxbytes
+.Cm s
+.Sm on
+.Pp
 The
-.Vt "void *"
-pointer argument is printed in hexadecimal (as if by
-.Ql %#x
-or
-.Ql %#lx ) .
-.It Cm n
-The number of characters written so far is stored into the
-integer indicated by the
-.Vt "int *"
-(or variant) pointer argument.
-No argument is converted.
-.It Cm m
-Print the string representation of the error code stored in the
-.Dv errno
-variable at the beginning of the call, as returned by
-.Xr strerror 3 .
-No argument is taken.
-.It Cm %
-A
-.Ql %
-is written.
-No argument is converted.
-The complete conversion specification
-is
-.Ql %% .
+.Vt "char *"
+argument is expected to be a pointer to an array of character type (pointer
+to a string).
+Characters from the array are written up to (but not including)
+a terminating
+.Dv NUL
+character;
+if a precision is specified, no more than the number specified are
+written.
+If a precision is given, no null character
+need be present; if the
+.Ar precision
+is not specified, or is greater than
+the size of the array, the array must contain a terminating
+.Dv NUL
+character.
+.It Cm U
+.Sm off
+.Cm %
+.Op Ar argno Cm $
+.Op Cm \~ | 0
+.Op .Ar mindigits
+.Op .Ar size
+.Cm U
+.Sm on
+.Pp
+A deprecated alias forg
+.Cm lu .
+.It Cm u
+.Sm off
+.Cm %
+.Op Ar argno Cm $
+.Op Cm \~ | 0
+.Op .Ar mindigits
+.Op .Ar size
+.Cm u
+.Sm on
+.Pp
+The
+.Vt unsigned int
+argument is converted to unsigned decimal notation.
+If specified, at least
+.Ar mindigits
+are printed, padding with leading zeros if needed.
+The following are similar to
+.Cm u
+except that they take an argument of a different size:
+.Bl -column %hhu
+.It Cm hhu Ta Vt unsigned char
+.It Cm hu  Ta Vt unsigned short
+.It Cm u   Ta Vt unsigned int
+.It Cm lu  Ta Vt unsigned long Pq percent ell u
+.It Cm llu Ta Vt unsigned long long Pq percent ell ell u
+.It Cm ju  Ta Vt uintmax_t
+.It Cm tu  Ta unsigned type of same size as Vt ptrdiff_t
+.It Cm zu  Ta Vt size_t
+.It Cm qu  Ta Vt u_quad_t Pq deprecated
 .El
 .Pp
-The decimal point
-character is defined in the program's locale (category
-.Dv LC_NUMERIC ) .
+.It Cm X
+.Sm off
+.Cm %
+.Op ar argno Cm $
+.Op Cm #
+.Op Cm \ | +
+.Op Cm \- | 0
+.Op Ar width
+.Op . Ar mindigits
+.Op Ar size
+.Cm X
+.Sm on
+.Pp
+Similar to
+.Vt x
+with the variation on the string value prefix '0x' to '0X'
```

Comment:
Ungrammatical and improper markup, but at least you got the prefixes right this time.